CredentialProvider.Microsoft/Util/HttpClientFactory.cs

```diff
@@ -23,11 +24,15 @@ static HttpClientFactory()
         {
             // https://learn.microsoft.com/en-us/dotnet/fundamentals/networking/http/httpclient-guidelines
 #if NETFRAMEWORK
-            var httpClient = new HttpClient();
+            var httpClient = new HttpClient(new HttpClientHandler
+            {
+                UseDefaultCredentials = true
+            });
 #else
             var httpClient  = new HttpClient(new SocketsHttpHandler
             {
-                PooledConnectionLifetime = TimeSpan.FromMinutes(15)
+                PooledConnectionLifetime = TimeSpan.FromMinutes(15),
+                DefaultProxyCredentials = CredentialCache.DefaultCredentials
```

Comment:
Not a great replacement for UseDefaultCredentials in SocketsHttpHandler it seems, may need to set Credentials here, though need more testing on that change. Can also just switch away from SocketsHttpHandler entirely which is what MSAL has done.

Reply:
I changed this to just match MSAL and confirmed it works.